### PR TITLE
[TEST] Refactor ImageData to be more strongly typed

### PR DESCRIPTION
**Pull request by MichaelSimons**
Refactor the ImageData class used in the tests.
- version properties are of type `Version`
- architecture is an enum
- compute sdkOs versus defining it
- shorten member names
- reformat test data declaration to be more readable
- misc cleanup

### CODE REVIEW

File: tests/Microsoft.DotNet.Docker.Tests/Dockerfile.linux.testapp [hunk 1]
```diff
@@ -19,6 +19,5 @@ RUN echo '<?xml version="1.0" encoding="utf-8"?>' >> NuGet.config \
     && echo '  </packageSources>' >> NuGet.config \
     && echo '</configuration>' >> NuGet.config \
     && echo '' >> NuGet.config
-RUN cat NuGet.config
```
Comment (dseefeld) [inline tests/Microsoft.DotNet.Docker.Tests/Dockerfile.linux.testapp]: Why was this here and why is it being removed?

Comment (MichaelSimons) [reply tests/Microsoft.DotNet.Docker.Tests/Dockerfile.linux.testapp]: It was there for debugging purposes and shouldn't have been checked in.

File: tests/Microsoft.DotNet.Docker.Tests/ImageVersion.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.DotNet.Docker.Tests
+{
+    public static class ImageVersion
```
Comment (dseefeld) [inline tests/Microsoft.DotNet.Docker.Tests/ImageVersion.cs]: What are these for?

Comment (MichaelSimons) [reply tests/Microsoft.DotNet.Docker.Tests/ImageVersion.cs]: There are used when defining the ImageData instances. I thought it was cleaner to define them here than inline in ImageTests.

Comment (MichaelSimons) [reply tests/Microsoft.DotNet.Docker.Tests/ImageVersion.cs]: See https://github.com/dotnet/dotnet-docker/blob/refactor-TestData/tests/Microsoft.DotNet.Docker.Tests/ImageTests.cs#L32
